### PR TITLE
On Linux restart if new version is found on disk

### LINKED ISSUE #3253: background linux packaging update does not notify end user in client
I have 1.8.0 running on opensuse 13.1 -- yast already installed 1.8.1 as online update in the background.

The 1.8.0 client keeps running and its General tab says 'no updates available'

I expect to see a notification saying e.g: 
  An update has been installed. Please restart your owncloud client to complete the update.

Workaround: logout or reboot frequently.

### PR DESCRIPTION
**Pull request by dragotin**
This will fix #3253 

@ogoffart, @ckamm please.

### CODE REVIEW

File: test/testutility.h [hunk 1]
```diff
@@ -103,6 +103,17 @@ private slots:
 
 
     }
+
+    void testVersionOfInstalledBinary()
+    {
+	if( isLinux() ) {
+                QString ver = versionOfInstalledBinary("/home/kf/owncloud.com/buildmirall/bin/owncloud");
```
Comment (ogoffart) [inline test/testutility.h]: this cannot work. 

Also it is only testing ver is not empty,  not that it is a version.

**Comment by guruz** [comment]
Do we really want this automatic?
It's a bit impolite

(i think @ogoffart wrote this too but i don't see his comment anymore)

CC @danimo

File: test/testutility.h (hunk 1 above)
Comment (dragotin) [inline test/testutility.h]: both fixed with subsequent commits.

**Comment by ogoffart** [comment]
I still think it is not wise to restart the app.

Anyway, regarding the patch,  is see no reason to put this in utility, since it is only used on one place from the GUI.
Also it only checks the first line,  but maybe the first line will be something like `XError on Display XYZ` for some reason. and this may lead to infinite restart.

We also do not put all the parametter.  Maybe the user put some -style or -geometry parameter in the .dekstop file, and when restarting this would not be applied.

We would also restart even if the user is doing something in the UI like adding a new folder.

**Comment by danimo** [comment]
I merged this for beta1, we can still refine/revert it if we do not like it.
